tests/integration_test/service_control_basic_test.go

```diff
@@ -201,33 +201,34 @@ func TestServiceControlBasic(t *testing.T) {
 			},
 		},
 		{
-			desc:     "succeed, no Jwt required, allow no api key (there is API key in request though it is not required), service control sends report only",
+			desc:     "SC does NOT check (but does report) when API Key is in the request, but operation does allow unregistered callers.",
 			url:      fmt.Sprintf("http://localhost:%v%v%v", s.Ports().ListenerPort, "/echo/nokey", "?key=api-key"),
 			message:  "hello",
 			method:   "POST",
 			wantResp: `{"message":"hello"}`,
 			wantScRequests: []interface{}{
 				&utils.ExpectedReport{
-					Version:                      utils.ESPv2Version(),
-					ServiceName:                  "echo-api.endpoints.cloudesf-testing.cloud.goog",
-					ServiceConfigID:              "test-config-id",
-					ApiKeyInOperationAndLogEntry: "api-key",
-					URL:                          "/echo/nokey?key=api-key",
-					ApiName:                      "1.echo_api_endpoints_cloudesf_testing_cloud_goog",
-					ApiMethod:                    "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Echo_nokey",
-					ProducerProjectID:            "producer-project",
-					HttpMethod:                   "POST",
-					FrontendProtocol:             "http",
-					LogMessage:                   "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Echo_nokey is called",
-					StatusCode:                   "0",
-					ResponseCode:                 200,
-					Platform:                     util.GCE,
-					Location:                     "test-zone",
+					Version:         utils.ESPv2Version(),
+					ServiceName:     "echo-api.endpoints.cloudesf-testing.cloud.goog",
+					ServiceConfigID: "test-config-id",
+					URL:             "/echo/nokey?key=api-key",
+					ApiName:         "1.echo_api_endpoints_cloudesf_testing_cloud_goog",
+					ApiMethod:       "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Echo_nokey",
+					// API Key is not verified by check, but still log it.
+					ApiKeyInLogEntryOnly: "api-key",
```

Comment:
I am wondering; will it be more readable if we rename them to 
ApiKeyVerified.   
ApiKeyNotVerified.

Reply:
I disagree. In our test framework, we should be validating how/where the API Key appears in the report request. It changes based on if it's verified or not, but that is implementation detail.

If we rename it to Verified/NotVerified, it hides the fact that verified keys appear in the operation or not. It will be very easy to forget when adding new tests.

Reply:
SG then.